### ISSUE TITLE
version 3.1.0 not work in Carthage

### ISSUE THREAD

**Issue by jhihguan** [LoganWright/Genome#75]
I'm trying Genome with Carthage, but when running `carthage update` it won't download `Packages`  folder content. Node, PathIndexable and Polymorphic is missing in project, which caused built failed.

I'm now running with specific version
`github "LoganWright/Genome" == 3.0.3`
It worked.

might be submodule issue?

**Comment by LoganWright** [comment]
@jhihguan I honestly have no idea 🙊 I'm not up and up on Carthage.

Would deleting the `.git` folders in Packages help perhaps?

**Comment by jhihguan** [comment]
I don't see `.git` in Packages folder.

When I clone the project in SourceTree it will error with
`fatal: No url found for submodule path 'Packages/Node-1.0.1' in .gitmodules`
and if I type `git submodule` in terminal, it will show
`fatal: no submodule mapping found in .gitmodules for path 'Packages/Node-1.0.1'`

But when I try `pod install` in another project, those package will download correctly.

yeah...I'm not up on Carthage😆

**Comment by max-potapov** [comment]
Got same git error with SPM:

```sh
$ swift build
Cloning https://github.com/LoganWright/Genome.git
/Users/max/.homebrew/bin/git clone --recursive --depth 10 https://github.com/LoganWright/Genome.git /Users/max/devel/***-swift/Packages/Genome.git
Cloning into '/Users/max/devel/***-swift/Packages/Genome.git'...
fatal: No url found for submodule path 'Packages/Node-1.0.1' in .gitmodules

error: Failed to clone https://github.com/LoganWright/Genome.git to /Users/max/devel/***-swift/Packages/Genome.git
```

**Comment by LoganWright** [comment]
@jhihguan @max-potapov can you attempt 3.1.1 and let me know if the issue is resolved.

**Comment by jhihguan** [comment]
I'm testing it in https://github.com/bwhiteley/JSONShootout
if I change version to 3.1.1 and do  `carthage update --no-build`
the`Genome.xcodeproj` still link to order version of `Node` `Polymorphic` `PathIndexable`
see the screenshot below
![screen shot 2016-11-08 at 14 36 44](https://cloud.githubusercontent.com/assets/745400/20089390/965891a8-a5c1-11e6-9a5c-005843c03ea9.png)

strange carthage...

**Comment by max-potapov** [comment]
@LoganWright tried 3.1.1 with spm and git recursive clone – everything is fine. thanks!

**Comment by jhihguan** [comment]
ahh, I think I found the issue, because carthage will use source code's `Genome.xcodeproj` to build project.
But right now `Node` `Polymorphic` `PathIndexable` is point to old code which is not existed anymore. (you can see my screenshot above)

I know how to add file into project, but I don't know how to link file like the way you did ex. `Core/Node+Init.swift`. Can you also tell me how you do that?

**Comment by LoganWright** [comment]
@jhihguan I was regenerating project from SPM, normally I wouldn't include the xcodeproj in the repository, but its necessary for Carthage. Let me try a regen and see what happens.

**Comment by LoganWright** [comment]
@jhihguan how does #79 look?

**Comment by LoganWright** [comment]
@jhihguan @max-potapov can you guys try 3.1.2 🙏

**Comment by jhihguan** [comment]
@LoganWright it works! 
maybe after @max-potapov confirmed than this issue can be closed

**Comment by jhihguan** [comment]
@LoganWright sorry... I found a small issue that 3.1.2 only build Mac device while 3.0.3 has Mac, iOS, watchOS, tvOS

**Comment by jhihguan** [comment]
I add platform support in #80 and also set the minimum version for each platform

**Comment by max-potapov** [comment]
@jhihguan @LoganWright both spm & carthage works fine with 3.1.2!

**Comment by LoganWright** [comment]
@jhihguan thanks, merged as 3.1.3

https://github.com/LoganWright/Genome/releases/tag/3.1.3

**Comment by jhihguan** [comment]
great! it works
